### PR TITLE
Add basic source-to-image integration tests

### PR DESCRIPTION
**Pull request by spadgett**
I'd like to add additional tests, but this is a start. It verifies the basic flow and will unblock @rhamilto who needs something similar for deploy image.

/cc @rhamilto @dtaylor113 @alecmerdler

### CODE REVIEW

File: frontend/integration-tests/tests/source-to-image.scenario.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    const appLabel = `app=${appName}`;
+    const resources = {
+      'buildconfigs': {kind: 'BuildConfig'},
+      'deploymentconfigs': {kind: 'DeploymentConfig'},
+      'imagestreams': {kind: 'ImageStream'},
+      'routes': {kind: 'Route'},
+      'services': {kind: 'Service'},
+    };
+
+    const visitResource = (resource: string) => {
```
Comment (alecmerdler) [inline frontend/integration-tests/tests/source-to-image.scenario.ts]: Should be an `async` function that returns a promise.

Comment (spadgett) [reply frontend/integration-tests/tests/source-to-image.scenario.ts]: I don't think `browser.get()` returns a promise

Comment (alecmerdler) [reply frontend/integration-tests/tests/source-to-image.scenario.ts]: It does ([source](https://github.com/angular/protractor/blob/master/lib/browser.ts#L859)) and we [usually write `await browser.get()`](https://github.com/openshift/console/blob/master/frontend/integration-tests/tests/crud.scenario.ts#L71).

Comment (spadgett) [reply frontend/integration-tests/tests/source-to-image.scenario.ts]: OK thanks, updated

File: frontend/integration-tests/views/source-to-image.view.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export const isLoaded = () => browser.wait(until.presenceOf($('.co-source-to-image-form')));
+export const nsDropdown = $('#namespace');
+export const nameInput = $('#name');
+export const trySampleButton = element(by.partialButtonText('Try Sample'));
+export const routeCheckbox = $('.checkbox input');
+export const submitButton = $('.btn-primary');
+export const errorMessage = $('.alert-danger');
+
+export const visitOpenShiftImageStream = async(name: string) => {
+  await browser.get(`${appHost}/k8s/ns/openshift/imagestreams/${name}`);
```
Comment (spadgett) [inline frontend/integration-tests/views/source-to-image.view.ts]: @alecmerdler added an `await` here too

**Pull request merged**